Hashing/hashMapWithDoubleHashing.cpp

```diff
@@ -0,0 +1,401 @@
+#include <bits/stdc++.h>
+using namespace std;
+using namespace std::chrono;
+
+// Creating HashMap with quadratic Probing
+
+// Double hashing is a collision resolution technique used in hash
+//  tables. It works by using two hash functions to compute two
+//  different hash values for a given key. The first hash function
+//  is used to compute the initial hash value, and the second hash
+//  function is used to compute the step size for the probing sequence.
+
+/*
+Double hashing can be done using :
+(hash1(key) + i * hash2(key)) % TABLE_SIZE
+Here hash1() and hash2() are hash functions and TABLE_SIZE
+is size of hash table.
+(We repeat by increasing i when collision occurs)
+*/
+
+/*
+
+A popular second hash function is
+hash2(key) = PRIME – (key % PRIME)
+where PRIME is a prime smaller than the TABLE_SIZE.
+
+*/
+
+// hash Node Class
+
+class HashNode
+{
+public:
+    int value;
+    int key;
+
+    HashNode(int key, int value)
+    {
+        this->key = key;
+        this->value = value;
+    }
+};
+
+// HashMap Class
+
+class HashMap
+{
+
+    // Hash elements array
+
+    HashNode **arr;
+
+    // hash capacity
+
+    int capacity;
+
+    // size
+    int size;
+
+    // counting collisions
+
+    int collisions;
+
+    // prime number less than the capacity
+
+    int prime;
+
+    // sieve of eratosthenes for marking all prime number less than the capcity
+
+    vector<bool> isPrime;
```

Comment:
Why to use Prime in Hash-function what is the concept behind it and what problem it is solving

Reply:
Using Prime reduces the chance of collision. By selecting prime just less than the bucket size reduces the count of collisions, since highly frequent number won't have prime number as its multiple.